src/tags/videotag.js

```diff
@@ -89,50 +104,69 @@ const VideoTag = class VideoTag extends HtmlTag {
-          srcType = sourceTypes[i];
-          transformation = sourceTransformation[srcType] || {};
-          src = url(`${this.publicId}`, defaults({}, transformation, {
-            resource_type: 'video',
-            format: srcType
-          }, options));
-          videoType = srcType === 'ogv' ? 'ogg' : srcType;
-          type = 'video/' + videoType;
-          results.push(`<source ${this.htmlAttrs({src, type})}>`);
-        }
-        return results;
-      }).call(this);
-    } else {
-      innerTags = [];
-    }
+    sourceTypes = this.transformation().getValue('source_types');
+    var innerTags = sources ?
+        this.handleSources(sources) :
+        this.handleSourceTypes(sourceTypes);
     return innerTags.join('') + fallback;
   }
 
+  handleSourceTypes(sourceTypes){
+    if(!isArray(sourceTypes)) return [];
+    const sourceTransformation = this.transformation().getValue('source_transformation');
+    let options = this.getOptions();
+    let results = [];
+    sourceTypes.forEach(srcType => {
+      let transformation = sourceTransformation[srcType] || {};
+      let src = url(`${this.publicId}`, defaults({}, transformation, {
+        resource_type: 'video',
+        format: srcType
+      }, options));
+      let type = 'video/' + (srcType === 'ogv' ? 'ogg' : srcType);
+      results.push(`<source ${this.htmlAttrs({src, type})}>`);
+    });
+    return results;
+  }
+
+  handleSources(sources){
+    var res = [];
+    let options = this.getOptions();
+    sources.forEach(currSrc =>{
+      let codecs = currSrc.codecs;
+      let type = 'video/'+currSrc.type;
+      if(codecs){
+        type +=  "; codecs="+codecs;
+      }
+      let src = url(`${this.publicId}`, defaults({}, currSrc.transformations, {
+        format: currSrc.type,
+        resource_type: 'video'
+      }, options));
+      res.push(`<source ${this.htmlAttrs({src, type})}>`);
+    });
+    return res;
+  }
+
   attributes() {
-    var a, attr, defaultOptions, i, len, poster, ref, ref1, sourceTypes;
-    sourceTypes = this.getOption('source_types');
-    poster = (ref = this.getOption('poster')) != null ? ref : {};
+    let defaultOptions, ref, ref1, i, len, a;
+    const sourceTypes = this.getOption('source_types');
+    let poster = (ref = this.getOption('poster')) != null ? ref : {};
+    const sources = this.getOption('sources');
     if (isPlainObject(poster)) {
       defaultOptions = poster.public_id != null ? DEFAULT_IMAGE_PARAMS : DEFAULT_POSTER_OPTIONS;
       poster = url((ref1 = poster.public_id) != null ? ref1 : this.publicId, defaults({}, poster, defaultOptions, this.getOptions()));
     }
-    attr = super.attributes() || [];
+    let attr = super.attributes() || [];
     for (i = 0, len = attr.length; i < len; i++) {
       a = attr[i];
       if (!contains(VIDEO_TAG_PARAMS)) {
         attr = a;
       }
     }
-    if (!isArray(sourceTypes)) {
+
+    if (!isArray(sourceTypes) && !sources) {
```

Comment:
This condition will change, considering above comment about handling empty `sources`array